src/installers.ts

```diff
@@ -459,6 +474,123 @@ export const installArchiveOnlyMod: VortexWrappedInstallFunc = (
   return Promise.resolve({ instructions });
 };
 
+export const testForJsonMod: VortexWrappedTestSupportedFunc = (
+  _api: VortexAPI,
+  log: VortexLogFunc,
+  files: string[],
+  gameId: string,
+): Promise<VortexTestResult> => {
+  // Make sure we're able to support this mod.
+  const correctGame = gameId === GAME_ID;
+  log("info", "Checking JSON files for game: ", gameId);
+  if (!correctGame) {
+    // Not in game mode?
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  let supported: boolean;
+  const filtered = files.filter(
+    (file: string) => path.extname(file).toLowerCase() === JSON_FILE_EXT,
+  );
+  if (filtered.length === 0) {
+    log("info", "No JSON files");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+
+  // just make sure we don't somehow have a CET mod that got here
+  const cetModJson = filtered.filter((file: string) =>
+    path.dirname(file).toLowerCase().includes(CET_MOD_CANONICAL_PATH_PREFIX),
+  );
+  if (cetModJson.length !== 0) {
+    log("error", "We somehow got a CET mod in the JSON check");
+    return Promise.resolve({
+      supported: false,
+      requiredFiles: [],
+    });
+  }
+  let proper = true;
+  // check for options.json in the file list
+  const options = filtered.some((file: string) =>
+    file.endsWith("options.json"),
+  );
+  if (options) {
+    log("debug", "Options.json files found: ", options);
+    proper = filtered.some((f: string) =>
+      path
+        .dirname(f)
+        .toLowerCase()
+        .startsWith(path.normalize("r6/config/settings")),
+    );
+
+    if (!proper) {
+      log(
+        "info",
+        "Improperly located options.json found in archive, we can't install this",
+      );
+      return Promise.reject(
+        new Error(
```

Comment:
This would probably be a nice spot to use a dialog! Both the notif and the dialog creation should be extracted, and having the second use case here will help get the abstraction right — don't have to do that in this PR if you just want to copy the logic for now.

And specifically I think just the dialog in this case? Because we really want the user to notice it (which isn’t guaranteed with a notification). I think you can pretty much just grab https://github.com/E1337Kat/cyberpunk2077_ext_redux/blob/master/src/installers.ts#L438 and use that part (IIRC the type is spelled out to "error", "warning" etc. unlike for logging.)

You can create a mock to make sure it gets called in a test case but there are no examples of that yet; either way you should be able to see it if you test manually in Vortex :)

Reply:
While I don't specifically call a dialog, the `Promise.reject(new Error(...) );` throws up a huge dialog letting the user know that it wasn't installed and giving the reason why (i.e. we don't know where their options.json goes).